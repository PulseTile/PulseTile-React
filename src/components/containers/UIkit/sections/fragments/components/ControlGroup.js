import React from 'react';

/**
 * This component returns content of ControlGroup section
 */
const ControlGroup = () => {
  return (
    <div id="control-group" className="ui-section">
      <strong className="ui-title">Control Group</strong>
      <div className="ui-sub-section">
        <strong className="ui-sub-title">Controls aligned to left</strong>
        <div className="wrap-control-group hide-indent-bottom">
          <div className="control-group with-indent left">
            <button className="btn btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <button className="btn btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
          </div>
        </div>
      </div>
      <div className="ui-sub-section">
        <strong className="ui-sub-title">Controls aligned to right</strong>
        <div className="wrap-control-group hide-indent-bottom">
          <div className="control-group with-indent right">
            <button className="btn btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <button className="btn btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
          </div>
        </div>
      </div>
      <div className="ui-sub-section">
        <strong className="ui-sub-title">Controls aligned to center</strong>
        <div className="wrap-control-group hide-indent-bottom">
          <div className="control-group with-indent center">
            <button className="btn btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <button className="btn btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
          </div>
        </div>
      </div>
      <div className="ui-sub-section">
        <strong className="ui-sub-title">Controls without indents</strong>
        <div className="wrap-control-group hide-indent-bottom">
          <div className="control-group with-indent without-side-indent left">
            <button className="btn btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <button className="btn btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
            <button className="btn btn-primary"><span className="btn-text">Button text</span></button>
          </div>
        </div>
      </div>
      <div className="ui-sub-section">
        <strong className="ui-sub-title">Controls with separates</strong>
        <div className="wrap-control-group">
          <div className="control-group with-indent left">
            <button className="btn btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate"></div>
            <button className="btn btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate"></div>
            <button className="btn btn-primary"><span className="btn-text">Button text</span></button>
          </div>
        </div>
        <div className="wrap-control-group">
          <div className="control-group with-indent left">
            <button className="btn btn-sm btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate control-separate-sm"></div>
            <button className="btn btn-sm btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate control-separate-sm"></div>
            <button className="btn btn-sm btn-primary"><span className="btn-text">Button text</span></button>
          </div>
        </div>
        <div className="wrap-control-group">
          <div className="control-group with-indent left">
            <button className="btn btn-smaller btn-danger"><i className="btn-icon fa fa-ban"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate control-separate-smaller"></div>
            <button className="btn btn-smaller btn-success"><i className="btn-icon fa fa-check"></i> <span className="btn-text">Button text</span></button>
            <div className="control-separate control-separate-smaller"></div>
            <button className="btn btn-smaller btn-primary"><span className="btn-text">Button text</span></button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ControlGroup;
