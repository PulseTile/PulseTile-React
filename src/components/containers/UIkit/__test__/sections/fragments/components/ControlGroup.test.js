import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import ControlGroup from '../../../../sections/fragments/components/ControlGroup';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <ControlGroup />', () => {
  it('should renders ControlGroup with props correctly', () => {
    const component = shallow(<ControlGroup />);
    expect(component).toMatchSnapshot();
  });
});
