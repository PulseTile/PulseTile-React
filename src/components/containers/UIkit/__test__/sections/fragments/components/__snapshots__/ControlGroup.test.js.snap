// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ControlGroup /> should renders ControlGroup with props correctly 1`] = `
<div
  className="ui-section"
  id="control-group"
>
  <strong
    className="ui-title"
  >
    Control Group
  </strong>
  <div
    className="ui-sub-section"
  >
    <strong
      className="ui-sub-title"
    >
      Controls aligned to left
    </strong>
    <div
      className="wrap-control-group hide-indent-bottom"
    >
      <div
        className="control-group with-indent left"
      >
        <button
          className="btn btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <button
          className="btn btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui-sub-section"
  >
    <strong
      className="ui-sub-title"
    >
      Controls aligned to right
    </strong>
    <div
      className="wrap-control-group hide-indent-bottom"
    >
      <div
        className="control-group with-indent right"
      >
        <button
          className="btn btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <button
          className="btn btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui-sub-section"
  >
    <strong
      className="ui-sub-title"
    >
      Controls aligned to center
    </strong>
    <div
      className="wrap-control-group hide-indent-bottom"
    >
      <div
        className="control-group with-indent center"
      >
        <button
          className="btn btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <button
          className="btn btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui-sub-section"
  >
    <strong
      className="ui-sub-title"
    >
      Controls without indents
    </strong>
    <div
      className="wrap-control-group hide-indent-bottom"
    >
      <div
        className="control-group with-indent without-side-indent left"
      >
        <button
          className="btn btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <button
          className="btn btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <button
          className="btn btn-primary"
        >
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui-sub-section"
  >
    <strong
      className="ui-sub-title"
    >
      Controls with separates
    </strong>
    <div
      className="wrap-control-group"
    >
      <div
        className="control-group with-indent left"
      >
        <button
          className="btn btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate"
        />
        <button
          className="btn btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate"
        />
        <button
          className="btn btn-primary"
        >
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
    <div
      className="wrap-control-group"
    >
      <div
        className="control-group with-indent left"
      >
        <button
          className="btn btn-sm btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate control-separate-sm"
        />
        <button
          className="btn btn-sm btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate control-separate-sm"
        />
        <button
          className="btn btn-sm btn-primary"
        >
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
    <div
      className="wrap-control-group"
    >
      <div
        className="control-group with-indent left"
      >
        <button
          className="btn btn-smaller btn-danger"
        >
          <i
            className="btn-icon fa fa-ban"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate control-separate-smaller"
        />
        <button
          className="btn btn-smaller btn-success"
        >
          <i
            className="btn-icon fa fa-check"
          />
           
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
        <div
          className="control-separate control-separate-smaller"
        />
        <button
          className="btn btn-smaller btn-primary"
        >
          <span
            className="btn-text"
          >
            Button text
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
